src/components/z-popover/index.tsx

```diff
@@ -208,125 +117,134 @@ export class ZPopover {
    * Close the popover when clicking outside of its content.
    * Stop event propagation if the click was fired by popover's trigger element,
    * to prevent close and reopen glitches.
-   * @param {MouseEvent} e
    */
   @Listen("click", {target: "body", capture: true})
   handleOutsideClick(e: MouseEvent): void {
-    if (!this.closable || !this.open) {
+    const target = e.target as Element;
+    if (!this.closable || !this.open || containsElement(this.host, target)) {
       return;
     }
 
-    const eventPath = e.composedPath();
-    if (!eventPath.includes(this.host)) {
-      const target = e.target as HTMLElement;
-      let triggerElemClicked = false;
-
-      if (typeof this.bindTo === "string") {
-        triggerElemClicked = !!target.closest(this.bindTo);
-      } else if (this.bindTo) {
-        triggerElemClicked = this.bindTo.contains(target);
-      }
-
-      if (triggerElemClicked) {
-        // stop propagation if the click was on the trigger element to prevent close and reopen glitches
-        e.stopPropagation();
-      }
-
-      this.open = false;
-      this.positionChange.emit({position: this.currentPosition});
+    if (containsElement(this.boundElement, target)) {
+      // stop propagation if the click was on the trigger element to prevent close and reopen glitches
+      e.stopPropagation();
     }
+    this.open = false;
   }
 
   @Watch("position")
   validatePosition(newValue: PopoverPosition): void {
     if (!Object.values(PopoverPosition).includes(newValue as PopoverPosition) || newValue === PopoverPosition.AUTO) {
       newValue = PopoverPosition.TOP;
+      this.position = newValue;
     }
-
-    this.position = newValue;
     this.currentPosition = newValue;
-    this.positionChange.emit({position: this.currentPosition});
-    this.setPosition();
+    if (this.open) {
+      this.setPosition();
+    }
   }
 
-  /**
-   * Setup popover behaviors on opening.
-   */
+  /** Setup popover behaviors when `open` changes. */
   @Watch("open")
   onOpen(): void {
     cancelAnimationFrame(this.animationFrameRequestId);
+    this.openChange.emit({open: this.open});
+    if (!this.open) {
+      return;
+    }
 
-    if (this.open) {
-      const setPosition = (): void => {
-        if (this.open) {
-          this.setPosition();
-          this.animationFrameRequestId = requestAnimationFrame(setPosition);
-        }
-      };
+    const setPosition = (): void => {
```

Comment:
Per chiarezza chiamerei questa funzione in modo diverso da setPosition ( che c'è già this.setPosition) magari qualcosa tipo `updatePositionLoop`